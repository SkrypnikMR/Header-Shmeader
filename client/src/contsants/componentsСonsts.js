export const regInputs = [{
    id: 'email',
    label: 'Email',
    placeholder: 'Enter your email...',
},
{
    id: 'password',
    label: 'Password',
    placeholder: 'Enter your password...',
},
{
    id: 'confirm',
    label: 'Confirm',
    placeholder: 'Confirm your password...',
},
{
    id: 'firstName',
    label: 'First name',
    placeholder: 'Enter your First name...',
},
{
    id: 'lastName',
    label: 'Last name',
    placeholder: 'Enter your Last name...',
}];

export const logInputs = [{
  id: 'email',
  label: 'Email',
  placeholder: 'Enter your email...',
},
{
  id: 'password',
  label: 'Password',
  placeholder: 'Enter your password...',
}];
