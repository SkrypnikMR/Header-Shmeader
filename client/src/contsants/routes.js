export const routes = {
    baseUrl: 'http://localhost:2282/api/',
    account: {
        registration: 'http://localhost:2282/api/account/registration',
        login: 'http://localhost:2282/api/account/login',
    },
};
