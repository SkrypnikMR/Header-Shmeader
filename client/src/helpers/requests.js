export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(url, options);
  return answer.json();
};
export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
