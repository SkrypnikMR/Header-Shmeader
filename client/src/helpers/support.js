export const support = {
    setSessionStorageItem: (name, data) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        sessionStorage.setItem(name, data);
    },
    killSessionStorageItem: (name) => {
        sessionStorage.removeItem(name);
    },
    getSessionStorageItem: (name) => {
        const item = sessionStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
    getPrettyTime: time => `${new Date(Number(time)).toTimeString().substr(0, 8)}`,
    getMessagesFolders: (rooms) => {
        const messagesFolders = {};
        rooms.forEach((room) => {
            const { room_name } = room;
            messagesFolders[room_name] = [];
        });
        return messagesFolders;
    },
    filteredRooms: (filterValue, rooms) => {
        const result = rooms.filter(room => room.room_name.includes(filterValue));
        return result.length === 0 ? null : result;
    },
};
