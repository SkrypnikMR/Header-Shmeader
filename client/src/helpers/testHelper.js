import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

export const shallowSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    const test = <Router>{core}</Router>;
    return shallow(test);
};
