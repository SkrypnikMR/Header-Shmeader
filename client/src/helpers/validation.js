import { regExp } from '../constants/regExp';

export const validation = {
    registrationValidation: ({ email, password, confirm, firstName, lastName }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        if (!regExp.firstNameRegExp.test(firstName) || !firstName) answer = { message: 'Invalid firstName', isValid: false };
        if (!regExp.lastNameRegExp.test(lastName) || !lastName) answer = { message: 'Invalid lastName', isValid: false };
        if (password !== confirm) answer = { message: 'Password mismatch', isValid: false };
        return answer;

    },
    loginValidation: ({ email, password }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        return answer;
    },
};
