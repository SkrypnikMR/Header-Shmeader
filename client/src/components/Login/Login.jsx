import React from 'react';
import PropTypes from 'prop-types';
import { logInputs } from '/src/contsants/componentsÐ¡onsts';
import { NavLink, Redirect } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import i18next from 'i18next';
import Form from '../UI/Form';
import Input from '../Ui/Input';
import Button from '../UI/Button';
import { APP_ROUTES } from '/src/contsants/reactRoutes';

const Login = ({ sendLoginRequest, setLoginValue, fields }) => {
  const { t } = useTranslation();
  if (fields.success) return <Redirect to={APP_ROUTES.chat} />;
  return (
    <Form>
      <p>{t('authorization')}</p>
      {logInputs.map(input => (
        <Input
          width="80%"
          id={input.id}
          height="100px"
          key={input.id}
          name={input.id}
          inputHeight="50px"
          borderRadius="5px"
          label={input.label}
          value={fields[input.id]}
          onChange={setLoginValue}
          placeholder={input.placeholder}
        />
      ))}
      <Button
        height="50px"
        content="Submit"
        id="login"
        onClick={sendLoginRequest}
      />
      <span>
        Need an account?
        <NavLink to={APP_ROUTES.registration}>Register now</NavLink>
      </span>
    </Form>
  );
};

Login.propTypes = {
  sendLoginRequest: PropTypes.func.isRequired,
  setLoginValue: PropTypes.func.isRequired,
  fields: PropTypes.object.isRequired,
};

export default Login;
