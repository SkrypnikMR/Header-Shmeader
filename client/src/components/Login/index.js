import { connect } from 'react-redux';
import Login from './Login.jsx';
import { setLoginValue, sendLoginRequest } from '/src/Store/login/actions';
import { loginStore } from '/src/Store/login/selectors';

const mapStateToProps = state => ({ fields: loginStore(state) });

const mapDispatchToProps = dispatch => ({
    setLoginValue: payload => dispatch(setLoginValue(payload)),
    sendLoginRequest: payload => dispatch(sendLoginRequest(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
