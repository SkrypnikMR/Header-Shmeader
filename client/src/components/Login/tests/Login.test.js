import React from 'react';
import { shallowSmart } from '/src/helpers/testHelper';
import { shallow } from 'enzyme';
import Login from '../Login.jsx';
import { logInputs } from '/src/contsants/componentsÐ¡onsts';

describe('Login', () => {
    let props;
    const sendLoginRequest = jest.fn();
    const setLoginValue = jest.fn();
    const fields = {
        email: '',
        password: '',
        success: null,
    };
    beforeEach(() => {
        props = {
            setLoginValue,
            sendLoginRequest,
            fields,
            logInputs,
        };
    });
    it('Should match snapshot', () => {
        const component = shallowSmart(<Login {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = shallow(<Login {...props} />);
        expect(component.find('p')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = shallow(<Login {...props} />);
        expect(component.find('Input')).toHaveLength(2);
    });
    it('should render button', () => {
        const component = shallow(<Login {...props} />);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should click on the button', () => {
        const component = shallow(<Login {...props} />);
        component.find('Button').simulate('click');
        expect(sendLoginRequest).toHaveBeenCalledWith();
    });
    it('should click change input', () => {
        const component = shallow(<Login {...props} />);
        component.find('Input').first().simulate('change', { name: 'email', value: 'emailValue' });
        expect(setLoginValue).toHaveBeenCalledWith({ name: 'email', value: 'emailValue' });
    });
});
