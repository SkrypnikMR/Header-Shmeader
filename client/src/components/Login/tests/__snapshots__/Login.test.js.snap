// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match snapshot 1`] = `"<div class=\\"styled__StLoginDiv-ohf2t6-0 cCvQFI\\"><div class=\\"styled__StFormDiv-sc-1xa63f-0 eesBZc\\"><p>مدخل</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"email\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">تحقق في</label><input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" placeholder=\\"...أدخل بريدك الإلكتروني\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"password\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">كلمه السر</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"...ادخل رقمك السري\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><button id=\\"login\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1a6hck8-0 biUGaX\\">ليأتي</button><span>احتاج الى حساب؟<a href=\\"/registration\\">انتقل إلى التسجيل</a></span></div></div>"`;
